### PR TITLE
snapraid: 11.1 -> 11.2

### PR DESCRIPTION
**Pull request by jfrankenau**
###### Motivation for this change

Update

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @makefu

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build snapraid

File: pkgs/tools/filesystems/snapraid/default.nix [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-    repo = "snapraid";
-    rev = "v${version}";
-    sha256 = "1vgyiagn6qvyyz4jkf8pzlg0w9apb6mzza97a9x03p82fav2afx0";
+  src = fetchgit {
+    url = "https://github.com/amadvance/snapraid";
+    rev = "refs/tags/v${version}";
+    sha256 = "05hq1478k61pa1hzg6xjaw63z6nba3yd7k2fxsxpb1fw06j5ywmj";
+    # needed by git for determining the version number
+    deepClone = true;
+    leaveDotGit = true;
```
Comment (Mic92) [inline pkgs/tools/filesystems/snapraid/default.nix]: Uh, this is always ugly and often break the checksum. Is it possible to do a simple patch instead?

File: pkgs/tools/filesystems/snapraid/default.nix [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
   };
```
Comment (jtojnar) [inline pkgs/tools/filesystems/snapraid/default.nix]: Alternately, you can use `.version` file: https://github.com/amadvance/snapraid/blob/f31dfada3e52e03487d4f68cfacb3b984381385b/autover.sh

```nix
postPatch = '''
  echo "${version}" > .version
''';
```

File: pkgs/tools/filesystems/snapraid/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/tools/filesystems/snapraid/default.nix]: You can simple say `VERSION=version;` in nix to make it a environment variable: https://github.com/amadvance/snapraid/blob/f31dfada3e52e03487d4f68cfacb3b984381385b/autover.sh#L14

Comment (jfrankenau) [reply pkgs/tools/filesystems/snapraid/default.nix]: You are right. This is a much better choice.

**Pull request merged**